### PR TITLE
[Server] Credential crud

### PR DESCRIPTION
**Pull request by Azanul**
**Notes for Reviewers**

This PR adds credential CRUD, same as meshery cloud.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Review by theBeginner86 — COMMENTED** [review]
Good stuff 👍

Few things to consider:
1. Credentials is a general capability of Meshery that is specific to the provider of the Meshery user. So logic of these handlers would go inside the [provider](https://github.com/meshery/meshery/blob/master/server/models/providers.go) interface. That is, these handlers' logic (in this PR) would be just necessary for [local provider](https://github.com/meshery/meshery/blob/master/server/models/default_local_provider.go)
2. There is a need to ensure that `Credentials` table is created when Meshery boots up (for local provider).
3. We need to expose credential as a capability of remote provider. _I tagged you in that recent PR where I exposed this_
4. Credentials is resource of group Integrations. So, the endpoint structure would be like: `/api/integrations/credentials`

To know more about how to ensure that handlers specific to the provider chosen by Meshery user, you can have a look at [other](https://github.com/meshery/meshery/blob/master/server/models/providers.go#L212) remote provider capabilities like `performance`, `patterns`, `applications`, `connections` etc.

Would you please make necessary changes?

**Comment by Azanul** [comment]
> Would you please make necessary changes?

Check it out

**Comment by leecalcote** [comment]
cha-cha-chickety-check it out, yo.

**Comment by Chadha93** [comment]
@Azanul  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by leecalcote** [comment]
@theBeginner86 and/or @MUzairS15 will need to review.

File: server/models/remote_provider.go [hunk 1]
```diff
@@ -3400,3 +3400,43 @@ func (l *RemoteProvider) SetKubeClient(client *mesherykube.Client) {
 func (l *RemoteProvider) GetKubeClient() *mesherykube.Client {
 	return l.KubeClient
 }
+
+// SaveCredential - to save a creadential for an integration
+// TODO
```
Comment (theBeginner86) [inline server/models/remote_provider.go:3405]: @Azanul 

Why is this remote_provider support left out? Are you about to add this?

Comment (Azanul) [reply server/models/remote_provider.go]: @theBeginner86 

There you go

**Pull request merged**